llvm/lib/Target/AArch64/GISel/AArch64InstructionSelector.cpp

```diff
@@ -3552,6 +3557,13 @@ bool AArch64InstructionSelector::select(MachineInstr &I) {
   return false;
 }
 
+bool AArch64InstructionSelector::selectAndRestoreState(MachineInstr &I) {
+  MachineIRBuilderState OldMIBState = MIB.getState();
+  bool Success = select(I);
+  MIB.getState() = std::move(OldMIBState);
```

Comment:
Just add a `setState()` instead? The state can be copied by value, it's just some pointers and an iterator.

Reply:
What about the DebugLoc? It has a TrackingMDRef, which does seem to benefit from moving, because it can call MetadataTracking::retrack(). I don't understand the details of MetadataTracking though.

Reply:
I'm also not familiar with that, but we already copy DebugLocs everywhere in llvm so I doubt think it's an issue?